### DISCUSSION TITLE
Testing "Proxy" experimental feature.

### DISCUSSION THREAD

**Discussion by denelon** [microsoft/winget-cli#4428] [Polls]
Please try the new experimental "Proxy" feature in WinGet 1.8 preview.

You can provide feedback in this discussion thread, but please do not add any additional comments to the Issue so we're not triggering a bunch of e-mail for folks who are subscribed there.

**Comment by soredake** [comment]
Can confirm that http proxy is working, would be nice to have support for socks5 proxies.

**Reply by denelon** [reply]
@soredake, feel free to create a new issue requesting support for socks5 proxies. It will largely depend on whether lots of folks request it or if someone from the community is willing to tackle it.

Thanks for the confirmation!!!

**Reply by pedoc** [reply]
Yes, it can be confirmed that ``http/s`` proxy works fine with ``--proxy`` option. 

Support for ``socks4/5`` is of course better 😊

**Comment by kingyue737** [comment]
How to use this feature？Add `"experimentalFeatures": {"proxy": true}` in settings and set `$Env:http_proxy`?

**Reply by soredake** [reply]
`--proxy "http://127.0.0.1:8118"` is one way to use proxy.

**Comment by 5night** [comment]
first,
`winget settings --enable ProxyCommandLineOptions`

then
`winget install --id Microsoft.Powershell --source winget --proxy http://127.0.0.1:7890`

**Comment by bennyyip** [comment]
How can I persist this option in a config file, instead of typing `--proxy http://127.0.0.1:7890` every time?

**Reply by florelis** [reply]
It can be persisted with the `settings` command:
```
> winget settings set DefaultProxy https://127.0.0.1:2345
> winget settings reset DefaultProxy
```

Note that this requires running as admin.

**Comment by msherian** [comment]
Unfortunately the only mechanism **supported** in my environment is a PAC file, and the current proxy implementation doesn't handle those.

**Reply by soredake** [reply]
You can open new feature request for this, as I did for SOCKS proxy support https://github.com/microsoft/winget-cli/issues/4465

**Comment by aith85** [comment]
What about proxy with user/pass authentication?

**Reply by tothambrus11** [reply]
This was my process:

- Download a recent version of winget from https://github.com/microsoft/winget-cli/releases (the msixbundle file)
- If needed, restart your machine so that the next step doesn't complain
- In admin powershell, run this: `Add-AppxPackage -Path "URL_TO_THE_MSIXBUNDLE_FILE"`
- Now you should have a recent winget version: `winget --version` -> v1.11.430
- Configure the proxy in an admin powershell: 
   - `winget settings --enable ProxyCommandLineOption`
   - `winget settings set DefaultProxy http://USERNAME:PASSWORD@proxy.mycompany.com:8080`
- Restart the terminal, then you should be good to go :D

**Comment by xiaoxiyao** [comment]
Can a configuration item be provided in the [settings](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md)?

**Reply by denelon** [reply]
Admin settings can specify a default proxy. There is an example in the [specification](https://github.com/microsoft/winget-cli/blob/master/doc/specs/%23190%20-%20Proxy%20Support.md).
